packages/salesforcedx-vscode-apex-debugger/src/adapter/apexDebug.ts

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) 2017, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {
+  DebugSession,
+  InitializedEvent,
+  OutputEvent
+} from 'vscode-debugadapter';
+import { DebugProtocol } from 'vscode-debugprotocol';
+import { SessionService } from '../core/sessionService';
+import { nls } from '../messages';
+import { CommandOutput } from '../utils/commandOutput';
+
+export interface LaunchRequestArguments
+  extends DebugProtocol.LaunchRequestArguments {
+  userIdFilter: string;
+  requestTypeFilter: string;
+  entryPointFilter: string;
+  sfdxProject: string;
+}
+
+export class ApexDebug extends DebugSession {
+  protected mySessionService = SessionService.getInstance();
+  private sfdxProject: string;
+
+  protected initializeRequest(
+    response: DebugProtocol.InitializeResponse,
+    args: DebugProtocol.InitializeRequestArguments
+  ): void {
+    this.sendResponse(response);
+    this.sendEvent(new InitializedEvent());
+  }
+
+  protected attachRequest(
+    response: DebugProtocol.AttachResponse,
+    args: DebugProtocol.AttachRequestArguments
+  ): void {
+    response.success = false;
+    this.sendResponse(response);
+  }
+
+  protected launchRequest(
+    response: DebugProtocol.LaunchResponse,
+    args: LaunchRequestArguments
+  ): void {
+    this.sfdxProject = args.sfdxProject;
+    this.mySessionService
+      .forProject(args.sfdxProject)
+      .withUserFilter(args.userIdFilter)
+      .withEntryFilter(args.entryPointFilter)
+      .withRequestFilter(args.requestTypeFilter)
+      .start()
+      .then(cmdResponse => {
```

Comment:
Do you have a preference for using `then`? I'm trying to change everything to use `await` so that we don't get into crazy then... hell.